core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/DataProcessor.scala

```diff
@@ -39,11 +39,10 @@ class DataProcessor( // dependencies:
         operator.open()
         runDPThreadMainLogic()
       } catch safely {
-        case e: InterruptedException =>
-          logger.logInfo("DP Thread exits")
+        case _: InterruptedException =>
+          logger.info("DP Thread exits")
         case e =>
-          val error = WorkflowRuntimeError(e, "DP Thread internal logic")
-          logger.logError(error)
+          logger.error(e.getLocalizedMessage + "\n" + e.getStackTrace.mkString("\n"))
```

Comment:
Should we send this error to the controller?

Reply:
done